### ISSUE TITLE
VTN not converging on UCF101

### ISSUE THREAD

**Issue by DAVEISHAN** [bomri/SlowFast#13]
Hello there, 

Thanks for sharing the code. I am not using the full codebase, instead, I am taking your model and fitting it into my training code. My training code is standard UCF101 classification code and works pretty well with the known 3D-CNN architectures. This issue can be very crucial for the people who are trying to use just your architecture for different video understanding tasks. I have done the following things, not sure what went wrong and VTN is not converging with my code:

1. I installed dependencies and was successfully able to build VTN model from [https://github.com/bomri/SlowFast/blob/master/slowfast/models/video_model_builder.py#L765](url), I used [https://github.com/bomri/SlowFast/blob/master/configs/Kinetics/VIT_B_VTN.yaml](url) config file and passed random input vector and vtn was able to produce some output of expected dimension without any error. 
2. Now, I changed the `NUM_CLASSES=101` for UCF101 in config yaml file and put the model into my training framework which has `adam optimizer` and dataloder that produces an output of 16 frames with skip rate of 2 and resolution of 224. The training loss slightly decreases from 4.8 --> 4.6 in an initial epoch and then just stuck there. 

I have no clue what has gone wrong, any lead would be appreciated. Hoping to hear from you soon. 

-Ishan

**Comment by DaniNem** [comment]
Hey Ishan,

Let's try to break this issue into smaller parts.
Few questions to start with:
1) Are you using the pre-trained kinetics model? (Transformers are very hard to train from scratch)
2) Did you try different lr's with adam? usually, the right lr zone is very narrow.
3) Did you try different optimizers like SGD? In our original paper, we trained with sgd, if you trained with kinetics pre-trained model it usually the sgd coverages

Thanks,
Daniel

**Comment by DAVEISHAN** [comment]
Hi Daniel, 

Thank you for your suggestions- I tried using SGD and it surprisingly works i.e. training loss started decreasing. Previously, when I was using ADAM optimizer, none of these settings worked: pretrained Kinetics/ImageNet, scratch with various learning rates from 1e-3 to 1e-4. 

Thanks again! Closing the issue :)

-Ishan
